src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/bolts/RerouteBolt.java

```diff
@@ -82,14 +87,18 @@ public void prepare(Map stormConf, TopologyContext context, OutputCollector coll
      */
     @Override
     protected void handleInput(Tuple tuple) throws PipelineException {
-        Message message = pullValue(tuple, FIELD_ID_PAYLOAD, Message.class);
-
-        if (message instanceof CommandMessage) {
-            handleCommandMessage((CommandMessage) message);
-        } else if (message instanceof InfoMessage) {
-            handleInfoMessage(message);
+        if (ZooKeeperSpout.BOLT_ID.equals(tuple.getSourceComponent())) {
+            LifecycleEvent event = (LifecycleEvent) tuple.getValueByField(FIELD_ID_LIFECYCLE_EVENT);
+            handleLifeCycleEvent(event);
         } else {
-            unhandledInput(tuple);
+            Message message = pullValue(tuple, FIELD_ID_PAYLOAD, Message.class);
+            if (active && message instanceof CommandMessage) {
+                handleCommandMessage((CommandMessage) message);
```

Comment:
I think we need here 
```
	   if (message instanceof CommandMessage) {
                if(active) {
                    handleCommandMessage((CommandMessage) message);
                 }
            } else if (message instanceof InfoMessage) {
                handleInfoMessage(message);
            } else {
                unhandledInput(tuple);
            }
```
To do not log unhandled tuples if topology is not active